store/pogreb/pogreb.go

```diff
@@ -161,37 +249,58 @@ func (s *pStorage) RemoveProvider(providerID peer.ID) error {
 	panic("not implemented")
 }
 
-// DuplicateEntry checks if the entry already exists in the index. An entry
-// for the same provider but different metadata is not considered
-// a duplicate entry,
-func duplicateEntry(in entry.Value, old []entry.Value) bool {
-	for i := range old {
-		if in.Equal(old[i]) {
-			return true
-		}
+func (s *pStorage) decreaseRefC(k []byte) (bool, error) {
+	old, found, err := s.getEntry(k)
+	if err != nil {
+		return false, err
+	}
+	if !found {
+		return false, errors.New("cannot decrease RefC from non-existing entry")
+	}
+
+	// Decrease RefC
+	old.RefC--
+	// If RefC == 0 remove the full entry
+	if old.RefC == 0 {
+		return true, s.entStore.Delete(k)
+	}
+	// If not we put entry with the updated refCount
+	b, err := store.Marshal(old)
+	if err != nil {
+		return false, err
+	}
+	s.entStore.Put(k, b)
+	if err != nil {
+		return false, err
 	}
-	return false
+	return true, nil
 }
 
-func (s *pStorage) removeEntry(k []byte, value entry.Value, stored []entry.Value) (bool, error) {
+func (s *pStorage) removeEntry(k []byte, value entry.Value, stored store.CidEntry) (bool, error) {
+	entK, err := store.EntryKey(value)
+	if err != nil {
+		return false, err
+	}
```

Comment:
If there are multiple concurrent calls to `removeEntry`, will this be a problem?